spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -169,11 +170,8 @@ private void doTaskStart() {
 				args = Arrays.asList(this.applicationArguments.getSourceArgs());
 			}
 
-			this.taskExecution = new TaskExecution(this.taskRepository.getNextExecutionId(),
-					null, this.taskNameResolver.getTaskName(), new Date(), null, null,
-					args);
-
-			this.taskRepository.createTaskExecution(this.taskExecution);
+			this.taskExecution = this.taskRepository.createTaskExecution(this.taskRepository.getNextExecutionId(),
```

Comment:
On the create, why are we providing the id?  Why is that not encapsulated within the create call?